### PR TITLE
OCLOMRS-415: Modify preview of convenient-set concept types to have set members, answers and other items as shown in the mock-ups

### PR DESCRIPTION
**Pull request by collinewait**
# JIRA TICKET NAME:
[Modify preview of convenient-set concept types to have set members, answers and other items as shown in the mock-ups](https://issues.openmrs.org/browse/OCLOMRS-415)

# Summary:
Any concept with set members should show the set members.   The same as true for any concept with coded answers.  These should show up when previewing a concept. Add items shown in the mocks-ups.

### CODE REVIEW

File: src/components/bulkConcepts/component/PreviewCard.jsx [hunk 1]
```diff
@@ -6,6 +6,25 @@ import {
 import CardBody from './CardBody';
 import MappingPreview from './MappingPreview';
 
+const getSetMembers = mappings => (
+  mappings.filter(mapping => mapping.map_type === 'CONCEPT-SET'))
```
Comment (dkayiwa) [inline src/components/bulkConcepts/component/PreviewCard.jsx]: Shouldn't you have constants for these?

File: src/components/bulkConcepts/component/PreviewCard.jsx [hunk 2]
```diff
@@ -16,60 +35,78 @@ const PreviewCard = ({
...
               <small>
-                <em>
-                  Version:&nbsp;
-                  {version}
-                </em>
-                <em>
-                  Created on:&nbsp;
-                  {new Date(created_on).toLocaleDateString('en-US', DATE_OPTIONS)}
-                </em>
+                <a href={`https://qa.openconceptlab.org${url}`} target="_blank" rel="noopener noreferrer">
```
Comment (dkayiwa) [inline src/components/bulkConcepts/component/PreviewCard.jsx]: I see the above url scattered around in the code base. Shouldn't we have a constant for it?

File: src/tests/bulkConcepts/components/PreviewCard.test.js [hunk 3]
```diff
@@ -48,6 +48,112 @@ describe('Test suite for ActionButton in BulkConceptsPage component', () => {
...
+      <PreviewCard {...newProps} />
+    </Router>);
+    expect(
+      wrapper.find('.synonyms').filterWhere(
+        node => node.text() === 'Answers',
+      ).length,
+    ).toEqual(1);
+  });
+
+  it('should call handleClick when more than five answers/set members are available', () => {
```
Comment (dkayiwa) [inline src/tests/bulkConcepts/components/PreviewCard.test.js]: Does this test mean that when more than five answers are available then handleClick is automatically called?

File: src/components/bulkConcepts/component/PreviewCard.jsx (hunk 1 above)
Comment (collinewait) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: Resolved. Constants have been used.

File: src/components/bulkConcepts/component/PreviewCard.jsx (hunk 2 above)
Comment (collinewait) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: I have created a constant for it.

File: src/tests/bulkConcepts/components/PreviewCard.test.js (hunk 3 above)
Comment (collinewait) [reply src/tests/bulkConcepts/components/PreviewCard.test.js]: Resolved. I have used a better description of the test.

File: src/components/bulkConcepts/component/PreviewCard.jsx [hunk 4]
```diff
@@ -16,60 +40,78 @@ const PreviewCard = ({
...
+  let remainingCount;
+  if (mapping !== 'none') {
+    setMembers = getSetMembers(mapping);
+    answers = getAnswers(mapping);
+    otherMappings = getOtherMappings(mapping);
+  }
+
+  const displaySetMembersOrAnswers = () => {
+    if (setMembers && setMembers.length > 0) {
+      remainingCount = setMembers.length - 5;
```
Comment (dkayiwa) [inline src/components/bulkConcepts/component/PreviewCard.jsx]: What does the magic number 5 stand for?

Comment (collinewait) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: We were required to display five items

Comment (dkayiwa) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: How would i tell that from the magic number 5?

Comment (collinewait) [reply src/components/bulkConcepts/component/PreviewCard.jsx]: I have fixed it.

**Pull request merged**